src/QsCompiler/CompilationManager/DataStructures.cs

```diff
@@ -41,69 +50,374 @@ internal class CodeLine
+                    if (inputDelimiter == "$\"")
+                    {
+                        errorDelimiterBuilder.Add(delim - 1); // Backup to the $ character
+                        delimiterBuilder.Add(delim);
+                    }
+                    else
+                    {
+                        errorDelimiterBuilder.Add(delim);
+                    }
+                }
+                else
+                {
+                    delimiterBuilder.Add(delim);
+                }
+            }
+
+            var lineLength = text.Length - this.LineEnding.Length;
+            // if there is a comment
+            if (commentStart >= 0)
+            {
+                this.WithoutEnding = text.Substring(0, commentStart);
+                var commentStr = text.Substring(commentStart, lineLength - commentStart).Trim();
+                var isDocComment = string.IsNullOrWhiteSpace(this.WithoutEnding) && (commentStr.Length - commentStr.TrimStart('/').Length == 3);
+                this.EndOfLineComment = isDocComment ? null : commentStr.Substring(2);
+            }
+            // else there is no comment
+            else
+            {
+                this.WithoutEnding = text.Substring(0, lineLength);
+                this.EndOfLineComment = null;
+            }
+
+            this.StringDelimiters = delimiterBuilder.ToImmutable();
+            this.ErrorDelimiterPositions = errorDelimiterBuilder.ToImmutable();
+            this.Indentation = indentation;
+            this.ExcessBracketPositions = excessBrackets.ToImmutableArray();
+            ScopeTracking.VerifyStringDelimiters(text, this.StringDelimiters);
+            ScopeTracking.VerifyExcessBracketPositions(this, excessBrackets);
         }
 
-        internal CodeLine SetStringDelimiters(IEnumerable<int> newStringDelimiters)
+        /// <summary>
+        /// Computes the location of the string delimiters within a given text.
+        /// </summary>
+        private static IEnumerable<int> ComputeStringDelimiters(string text, ref StringContext stringContext, out int commentIndex, out IEnumerable<int> errorDelimiters)
         {
-            return new CodeLine(this.Text, newStringDelimiters, this.WithoutEnding.Length, this.Indentation, this.ExcessBracketPositions);
+            var delimiterBuilder = ImmutableArray.CreateBuilder<int>();
+            var errorDelimiterBuilder = ImmutableArray.CreateBuilder<int>();
+
+            if (stringContext >= StringContext.OpenString)
+            {
+                delimiterBuilder.Add(-1);
+            }
+
+            commentIndex = -1;
+            var stringLength = text.Length;
+            while (text != string.Empty)
+            {
+                // Check for a comment start if we are outside a string
+                commentIndex = stringContext >= StringContext.OpenString ? -1 : text.IndexOf("//");
```

Comment:
Suggest making this check a separate function as it has a meaning.
stringContext >= StringContext.OpenString

Reply:
This is now encapsulated in the IsInAString function.